### ISSUE TITLE
keyboard shortcut to jump to next/previous line with the same indentation level

### ISSUE THREAD

**Issue by hprodh** [spyder-ide/spyder#22133]
I have a feature/enhancement request for the editor.
I would like to navigate faster within code using the keyboard (typically from class to class or function to function, etc...).
One perfect feature would be to have a keyboard shortcut to make the cursor jump to the next or previous line having the same indentation level as the current one. So that the shortcut would allow to easily jump indented blocks.
By the way, another shortcut to go the previous/next line having one less indentation level would be very nice additionally.

**Comment by dalthviz** [comment]
Hi @hprodh thank you for the feedback! Maybe something that could be useful for you then is the symbol switcher? With it, you can jump between definitions (functions, classes, methods, etc):

![symbol_switcher](https://github.com/user-attachments/assets/732f4f5b-568e-40de-a106-076cf98aa3cb)

To show the symbol switcher you can use the Ctrl +Alt + P shortcut.

Let us know if the info above helps!

**Comment by hprodh** [comment]
Hi @dalthviz,
I just tried this functionality, this looks very nice indeed, usable from keyboard only !

Yet I find one problem with the use of the symbol finder : when opening it with Ctrl+Alt+P, the editor cursor goes to the first symbol. This throws back to the beginning of the file. 
It would be very practical if it could open with the directly previous symbol selected, (and possibly should not move the cursor at opening).

**Comment by dalthviz** [comment]
> It would be very practical if it could open with the directly previous symbol selected

Thank you @hprodh for the extra feedback! I would say that makes sense to me :+1: Also, thinking about not changing the line where you are, I remembered that you can also activate a selector for classes/functions by checking `Source > Show selector for classes and functions`:

![symbol_selector](https://github.com/user-attachments/assets/f9fe267b-4442-42d1-aa1e-5f334e7b50f2)

Anyhow, in general, what do you think about these ideas @spyder-ide/core-developers (shortcuts to navigate the next/previous symbol and improvements to the symbol finder/switcher) ?

**Comment by hprodh** [comment]
Yes, this selector is very close to what the symbol switcher is, yet they each have a drawback for "fast-keyboard-use-only" :
- symbol switcher sends the cursor to the first symbol of the file at opening.
- selector requires the usage of the mouse to open it (also, it takes a bit of space on screen).

**Comment by hprodh** [comment]
Also, issue #910 is somehow related

**Comment by dalthviz** [comment]
Thank you for giving the selector a try @hprodh !

> Additionally, a simple "bookmark" comment, if recognized as a symbol, would maybe close the issue https://github.com/spyder-ide/spyder/issues/910. 

Just in case, there is already a mark/comment you can use to go to a specific place in your code. To use that functionality you just need to write `# --- <some text>` So for example, inside a class, if you want to mark a "section" where public methods are defined you can do something like `# ---- Public API`. After doing that you should see something like:

![image](https://github.com/user-attachments/assets/02e5e42a-4caf-4a8a-92c1-ed3403d1a756)

Those mark comments work for the symbol finder and also for the Outline pane

**Comment by hprodh** [comment]
I have opened the new issue #23138. I am closing this one.
